xsecurelock: Add mplayer/htpasswd support

SevereOverfl0w opened the pull request:
Add as build dependencies allows the makefile to detect mpv/mplayer
location, and will then include them as optional runtime savers.

xtraeme commented on srcpkgs/xsecurelock/template:
```diff
@@ -1,13 +1,14 @@
  libXft-devel libXmu-devel libXrandr-devel libXScrnSaver-devel
- libXxf86misc-devel pam-devel"
+ libXxf86misc-devel pam-devel
+ mpv mplayer"
```
Those belong to `hostmakedepends` instead

Duncaen commented:
Can this be done with `-with-mpv=/usr/bin/mpv` instead of adding extra `hostmakedepends`?

xtraeme commented:
right... aka autoconf cache

xtraeme commented:
We've been using this for maybe 15 years or even more in pkgsrc... xbps-src is millenial stuff

SevereOverfl0w commented:
I didn't realize that it wasn't actually checking for file support.  I've switched to using the --with, and also added htpasswd because that one was obvious.  xscreensaver was not obvious to me, so I'll leave that to someone else who has the desire to get that working.

sgn commented on srcpkgs/xsecurelock/template:
```diff
@@ -1,9 +1,9 @@
+revision=2
 build_style=gnu-configure
-configure_args="--with-pam-service-name=system-local-login"
+configure_args="--with-pam-service-name=system-local-login --with-mpv=/usr/bin/mpv --with-mplayer=/usr/bin/mplayer --with-htpasswd=/usr/bin/htpasswd"
```
too long line, please wrap it.

SevereOverfl0w commented:
I've updated the template to:

- No longer depend on mpv for mpv support
- Support mplayer & htpasswd
